lib/core/settle.js

```diff
@@ -11,7 +11,7 @@ var createError = require('./createError');
  */
 module.exports = function settle(resolve, reject, response) {
   var validateStatus = response.config.validateStatus;
-  if (!validateStatus || validateStatus(response.status)) {
+  if (validateStatus === null || (utils.isFunction(validateStatus) && validateStatus(response.status))) {
```

Comment:
we don't want to check specifically for `null`, if `validateStatus` is not passed, the promise should resolve

Reply:
> if `validateStatus` is not passed, the promise should resolve

should resolve? I think the promise should rejecte...

Reply:
how come? if I do:
```javascript
const config = { baseUrl: '', ... };
const client = axios(config) // config without validateStatus callback
client.get('/').then(() => ...)
```
I'm almost positive that all requests from that client would be rejected regardless of what's passed in the config if it doesn't have a `validateStatus`, wouldn't this change the default behavior of having vs not having validateStatus? Unless I'm missing something here

Reply:
When `validateStatus` is not been set, then the `validateStatus` is a function:
https://github.com/axios/axios/blob/841466416b6851666955113a60ae46830a27003f/lib/defaults.js#L78-L80

if someone set `validateStatus` to `true` or `1`, then `validateStatus(response.status)` will report an error.
https://github.com/axios/axios/blob/841466416b6851666955113a60ae46830a27003f/lib/core/settle.js#L14-L24

so I check whether the `validateStatus` is function first, if it is , then to judge the return value. What do you think?

Reply:
if you approve of this, then I remove the `null` condition and testcase